### ISSUE TITLE
SetuptoolsDeprecationWarning: setup.py install is deprecated

### ISSUE THREAD

**Issue by fenugrec** [pR0Ps/pycrk#2]
I'm trying to package this for Archlinux; I'm checking out the repo and then running `python setup.py build`  :

`SetuptoolsDeprecationWarning: setup.py install is deprecated. Use build and pip and other standards-based tools.`

As I understand, that is related to https://peps.python.org/pep-0517/ 

Not sure how to fix this.

Possibly related (or maybe just my incompetence), my build is broken :
```

$ crk-generate 
Traceback (most recent call last):
  File "/usr/bin/crk-generate", line 33, in <module>
    sys.exit(load_entry_point('pycrk==0.0.1', 'console_scripts', 'crk-generate')())
  File "/usr/bin/crk-generate", line 25, in importlib_load_entry_point
    return next(matches).load()
  File "/usr/lib/python3.10/importlib/metadata/__init__.py", line 171, in load
    module = import_module(match.group('module'))
  File "/usr/lib/python3.10/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1050, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1027, in _find_and_load
  File "<frozen importlib._bootstrap>", line 992, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
  File "<frozen importlib._bootstrap>", line 1050, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1027, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1006, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 688, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 883, in exec_module
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
  File "/usr/lib/python3.10/site-packages/pycrk/__init__.py", line 31, in <module>
    def _find_changes(original: os.PathLike, patched: os.PathLike) -> List[Change]:
NameError: name 'Change' is not defined. Did you mean: 'range'?

```

**Comment by pR0Ps** [comment]
I fixed #3 so it should build fine now. I think the deprecation that message is referring to is using `setup.py` itself to do the building.

The following method to create packages worked for me:
```bash
$ pip install build
<snip>
$ python -m build
<snip>
Successfully built pycrk-0.0.1.tar.gz and pycrk-0.0.1-py3-none-any.whl
```

Let me know if that works for you. I'll put out a new release with the fix and a git tag and everything soon.

**Comment by fenugrec** [comment]
Ok, haven't tried your method yet, I've settled with `python -m build --wheel --no-isolation`  , and this crude port to pyproject.toml + setup.cfg : 
https://github.com/fenugrec/pycrk/commit/cbd44ae8282b41b5dbdb30039888827e625efe8c

**Comment by pR0Ps** [comment]
I just pushed and tagged a proper v1.0.0. Future releases will also get a tag to make updates easier.

I just tested your `python -m build --wheel --no-isolation` command it built everything even with the current `setup.py`. I'll eventually migrate to the new `pyproject.toml` setup, but at the moment I'd prefer to take the conservative approach and stick with the `setup.py` while `pyproject.toml` gets more widely adopted.
